timeline/models.py

```diff
@@ -46,6 +47,22 @@ def get_object_events(
             event_objects__object_uuid=object_uuid,
         ).order_by(order_by)
 
+    def find(self, search_terms, keywords=None):
+        """
+        Return events matching the query.
+
+        :param search_terms: Search terms (list of strings)
+        :param keywords: Optional search keywords as key/value pairs (dict)
+        :return: QuerySet of ProjectEvent objects
+        """
+        objects = super().get_queryset().order_by('event_name')
+        term_query = Q()
+        for t in search_terms:
+            term_query.add(Q(event_name__icontains=t), Q.OR)
+            term_query.add(Q(description__icontains=t), Q.OR)
+            term_query.add(Q(project__full_title__icontains=t), Q.OR)
```

Comment:
Searching by the project full title was in my original spec. Now, after testing the feature for a bit, I realize I was wrong to design it that way :)

The project title search should be removed, as it just returns a ton of elements for each project matching the substring, which will not be very useful. One can simply browse the project timeline for that.